context.go

```diff
@@ -9,16 +9,21 @@ import (
 
 type clientsetCtxKey struct{}
 
-func setClientsetOnContext(ctx context.Context, clientset *kubernetes.Clientset) context.Context {
+// Note: we do not use kubernetes.Clientset type directly, to enable mocking via "k8s.io/client-go/kubernetes/fake"
+// package. A type alias is introduced to avoid confusion: "kubernetes.Interface" is actually a clientset.Interface, and
+// not a generic interface for Kubernetes resources.
```

Comment:
Added clarification and type aliasing for better readability (`kubernetes.Interface` is confusing)